Source/WebCore/Modules/indexeddb/server/MemoryIndexCursor.cpp

```diff
@@ -40,22 +40,27 @@ namespace IDBServer {
 
 WTF_MAKE_TZONE_ALLOCATED_IMPL(MemoryIndexCursor);
 
-MemoryIndexCursor::MemoryIndexCursor(MemoryIndex& index, const IDBCursorInfo& info)
-    : MemoryCursor(info)
+Ref<MemoryIndexCursor> MemoryIndexCursor::create(MemoryIndex& index, const IDBCursorInfo& info, MemoryBackingStoreTransaction& transaction)
+{
+    return adoptRef(*new MemoryIndexCursor(index, info, transaction));
+}
+
+MemoryIndexCursor::MemoryIndexCursor(MemoryIndex& index, const IDBCursorInfo& cursorInfo, MemoryBackingStoreTransaction& transaction)
+    : MemoryCursor(cursorInfo, transaction)
     , m_index(index)
 {
-    LOG(IndexedDB, "MemoryIndexCursor::MemoryIndexCursor %s", info.range().loggingString().utf8().data());
+    LOG(IndexedDB, "MemoryIndexCursor::MemoryIndexCursor %s", cursorInfo.range().loggingString().utf8().data());
 
     auto* valueStore = index.valueStore();
     if (!valueStore)
         return;
 
-    if (m_info.isDirectionForward())
-        m_currentIterator = valueStore->find(m_info.range().lowerKey, m_info.range().lowerOpen);
+    if (info().isDirectionForward())
```

Comment:
Perhaps we could make a local variable that stores the result of `info()` and use that instead of calling `info()` in many places? (Same for the other functions that have multiple calls to `info()`).

Reply:
Not a bad idea; perhaps we could do some cleanup in next patches.